runtime/src/asset.rs

```diff
@@ -89,13 +90,48 @@ pub trait Trait:
     type Currency: Currency<Self::AccountId>;
 }
 
+/// The type of an asset represented by a token.
+#[derive(codec::Encode, codec::Decode, Clone, PartialEq, Debug)]
+pub enum AssetType {
+    Equity,
+    Debt,
+    Commodity,
+    StructuredProduct,
+    Custom(Vec<u8>),
+}
+
+impl Default for AssetType {
+    fn default() -> Self {
+        AssetType::Custom(b"undefined".to_vec())
+    }
+}
+
+/// The type of an identifier associated with a token.
+#[derive(codec::Encode, codec::Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum IdentifierType {
+    Isin,
+    Cusip,
+    Custom(Vec<u8>),
+}
+
+impl Default for IdentifierType {
+    fn default() -> Self {
+        IdentifierType::Custom(b"undefined".to_vec())
+    }
+}
+
+/// A notational shorthand for the identifier map type.
+pub type Identifiers = BTreeMap<IdentifierType, Vec<u8>>;
```

Comment:
I don't think we need to store it as a tree. We probably won't be doing anything with this information onchain. A plain old vector will work just fine here.

Reply:
A vector allows to store duplicate identifiers. A map takes care of duplicates, which is a requirement.

Reply:
The reason I'm afraid to store it as a BTreeMap is that I am not sure how polkadot.js will handle it. Can you please try creating a token from polkadot.js and reading the data? If it works, then this is ok.

Otherwise, we can create a new map for identifiers like `map (ticker, identifier) -> Vec<u8>` and take input in an array and store it in the map. It will give us similar results and will definitely work with polkadot.js.

Reply:
The [changelog](https://changelogs.md/github/polkadot-js/api/) states support of `BTreeMap`.

Reply:
Everything works in polkadot.js but the `"identifiers": "BTreeMap<IdentifierType, Vec<u8>>"` field of `SecurityToken`. The enums `AssetType` and `IdentityType` work.